File: _0_language/_concurrency/_2_thread_lifecycle_and_management/_threads_management/thread.py
import threading
import time

def print_message():
    for i in range(5):
        print(f"Thread running: {i}")
        time.sleep(1)  # Simulate some work

# Create and start the thread
thread = threading.Thread(target=print_message)
thread.start()

# Wait for the thread to finish
thread.join()

print("Main thread ends.")

File: _0_language/_concurrency/_2_thread_lifecycle_and_management/_threads_management/thread.go
package main

import (
    "fmt"
    "time"
    "sync"
)

func printMessage(wg *sync.WaitGroup) {
    defer wg.Done()
    for i := 0; i < 5; i++ {
        fmt.Println("Goroutine running:", i)
        time.Sleep(1 * time.Second)
    }
}

func main() {
    var wg sync.WaitGroup
    wg.Add(1)  // Add 1 goroutine to the WaitGroup

    // Start the goroutine
    go printMessage(&wg)

    // Wait for the goroutine to finish
    wg.Wait()

    fmt.Println("Main goroutine ends.")
}

// In Go, we use a WaitGroup to wait for a goroutine to finish before the main function exits. 
// Goroutines are a more lightweight form of thread, and the sync package helps with managing their execution.